src/test/java/TestClass.java

```diff
@@ -0,0 +1,241 @@
+
+            if (res == 0) {
+                intervals.add(interval);
+                interval.setLastUpdateDateTime(Instant.now());
+                return true;
+            }
+
+            return false;
+        });
+
+        when(intervalsWorkerMock.setIntervalProcessed(any(Interval.class), anyBoolean())).then(invocation -> {
+            Interval interval = invocation.getArgument(0);
+            boolean isProcessed = invocation.getArgument(1);
+
+            List<Interval> intervalList = intervals.stream().filter(i -> i.getStartTime().equals(interval.getStartTime())
+                    && i.getCrawlerName().equals(interval.getCrawlerName())
+                    && i.getCrawlerVersion().equals(interval.getCrawlerVersion())
+                    && i.getCrawlerType().equals(interval.getCrawlerType())).collect(Collectors.toList());
+
+            Interval storedInterval = intervalList.get(0);
+
+            if (!storedInterval.getLastUpdateDateTime().equals(interval.getLastUpdateDateTime()))
+                throw new UpdateNotAppliedException("Failed to set processed flag to "+isProcessed+" at interval from "
+                        +interval.getStartTime()+", to "+interval.getEndTime());
+
+            interval.setProcessed(isProcessed);
+            interval.setLastUpdateDateTime(Instant.now());
+
+            return interval;
+        });
+
+        crawler = new Crawler(storageMock, dataServiceMock, dataProviderMock, configuration, new CrawlerTimeTestImpl());
+    }
+
+    private List<StreamResponse> addEvents() {
+
+        List<StreamResponse> responses = new ArrayList<>();
+
+        for (int i = 0; i < 100; i++) {
+            StreamResponse response = StreamResponse.newBuilder()
+                    .setEvent(EventData.newBuilder()
+                            .setEventId(toEventID(i +"id"))
+                            .setEventName("0.0.1|SearchProviderDataTest|"+Instant.now())
+                            .setStartTimestamp(toTimestamp(Instant.parse(configuration.getFrom()).plus(i * 10, ChronoUnit.MINUTES)))
+                            .build())
+                    .build();
+
+            responses.add(response);
+        }
+
+        return responses;
+    }
+
+    @Test
+    @DisplayName("Requiring handshake, getting the same name and version")
+    public void handshakeNeededSame() throws IOException, UnexpectedDataServiceException {
+
+        when(dataServiceMock.sendEvent(any(EventDataRequest.class))).then(invocation -> {
+            EventDataRequest request = invocation.getArgument(0);
+
+            List<EventData> events = request.getEventDataList();
+
+            EventID eventID = events.get(events.size() - 1).getEventId();
+
+            return EventResponse.newBuilder().setId(eventID).setStatus(Status.newBuilder().setHandshakeRequired(true).build()).build();
+        });
+
+        crawler.process();
+
+        // TODO: how to check logger message here?
```

Comment:
You cannot check the logger message. If information goes only into the logger you probably don't need to check it. It is only information for a user and a developer.